splot/_viz_esda_mpl.py

```diff
@@ -42,13 +44,111 @@ def _create_moran_fig_ax(ax, figsize):
     ax.spines['top'].set_color('none')
     ax.spines['left'].set_smart_bounds(True)
     ax.spines['bottom'].set_smart_bounds(True)
+    return fig, ax
+
+
+def moran_scatterplot(moran, zstandard=True, p=None, ax=None,
+                      scatter_kwds=None, fitline_kwds=None):
+    """
+    Moran Scatterplot
+    
+    Parameters
+    ----------
+    moran : esda.moran instance
+        Values of Moran's I Global, Bivariate and Local Autocorrelation Statistics
+    zstandard : bool, optional
+        If True, Moran Scatterplot will show z-standardized attribute and
+        spatial lag values. Default =True.
+    p : float, optional
+        If given, the p-value threshold for significance
+        for Local Autocorrelation analysis. Points will be colored by
+        significance. By default it will not be colored.
+        Default =None.
+    ax : Matplotlib Axes instance, optional
+        If given, the Moran plot will be created inside this axis.
+        Default =None.
+    scatter_kwds : keyword arguments, optional
+        Keywords used for creating and designing the scatter points.
+        Default =None.
+    fitline_kwds : keyword arguments, optional
+        Keywords used for creating and designing the moran fitline.
+        Default =None.
+
+    Returns
+    -------
+    fig : Matplotlib Figure instance
```

Comment:
Does it always return a `fig` object? Even when `ax` is not `None`? I couldn't tell, but usually, if you pass and `ax` I think usually you return only `ax` (not completely sure though).

Reply:
yes, it always returns a `fig`. If `ax` is not `None`, `else: fig = ax.get_figure()` will create a `fig`.